### DISCUSSION TITLE
Property Names - Something other than Camel/Pacal Case

### DISCUSSION THREAD

**Discussion by bo55vxr** [ChilliCream/hotchocolate#5194] [Q&A] [answered]
I am redeveloping an existing GraphQL service and need to replicate it exactly as any client applications cannot be affected by the underlying change.

Everything is going well, but I have one very specific issue in that when Hot Chocolate returns the data to the consumer, the JSon attribute names all start with a lower case letter. Now this is not a problem, except on ONE specific attribute. 

The GraphQL request indicates that the field to be requested is `Rating` but HC is exposing as `rating` so then HC throws an error stating is cannot find the field `Rating`. If I change the query to request `rating` then it works.

As I need to explicitly expose `Rating`, how can I do this... for this one field. I have tried using `[JsonPropertyName('Rating')]` attribute but this is not working.

Thanks in advance

**Comment by bo55vxr** [comment, marked as answer]
I'll answer my own quesion... A _Resolver_ and setting the _Name_ property

```
protected override void Configure(IObjectTypeDescriptor<Models.Product> descriptor)
{
    descriptor.Field(fld => fld.Rating).Name("Rating");
}
```

**Reply by ghost** [reply]
Or just use ```[GraphQLName("Rating")]``` on the property rather than ```JsonPropertyName```?

**Reply by bo55vxr** [reply]
> Or just use `[GraphQLName("Rating")]` on the property rather than `JsonPropertyName`?

Oh my ... how simple!
